subprojects/packagefiles/sdl2/meson.build

```diff
@@ -367,11 +368,28 @@ endif
 core_deps = [
     cc.find_library('m', required : false),
     cc.find_library('dl', required : false),
-    cc.find_library('iconv', required : false),
     dbus_dep,
     ibus_dep,
 ]
 
+if not opt_system_iconv.disabled()
+    if meson.version().version_compare('>=0.60')
+        iconv_dep = dependency('iconv', required : opt_system_iconv)
+    else
+        iconv_dep = cc.find_library('iconv', required : opt_system_iconv)
+    endif
+else
+    iconv_dep = fake_dep
+endif
+
+if cc.links('#include <iconv.h>\nint main() { iconv_open("", ""); }', name : 'iconv_open test', dependencies : iconv_dep)
+    cdata.set10('HAVE_ICONV', true)
+    cdata.set10('HAVE_SYSTEM_ICONV', true)
```

Comment:
If iconv is disabled this check still runs, and may find it as a built-in and set these defines to enable its use.

It should only run inside the `if not opt_system_iconv.disabled()` block.

Reply:
This option is not to disable iconv but to disable relying on an iconv library.

When the option is disabled it will check if libc provides it and otherwise disable iconv support.

Since we have features in meson the idea is
`auto` = lib if available otherwise libc
`enabled` = lib only
`disabled` = libc only

`dependency('iconv')` is useful, but will overall require more logic to handle the above.
Does the dependency solution do anything we cannot recreate using `find_library` and `links` ?

Reply:
That seems like a strange option. Why would someone want to make that decision?

"automatically enable iconv functionality if it's a builtin, otherwise allow the user to choose"?

Reply:
This copies the upstream behavior for enabled/disabled but adds extra functionality for auto which checks for both in the order of library and then libc.

Reply:
Also some more context:
this is wanted for Lite-XL, where Windows builds currently link against libiconv because of SDL, even if thats not what is wanted.
The point is not to add an iconv toggle.

Reply:
would it be better if I simply replaced it with a boolean and made it behave like upstream?

Reply:
bump

Reply:
one last bump before I consider this PR stale

Reply:
>  this is wanted for Lite-XL, where Windows builds currently link against libiconv because of SDL, even if thats not what is wanted.

So... lite-xl just wants to turn iconv off unconditionally? Am I understanding correctly?

Reply:
> So... lite-xl just wants to turn iconv off unconditionally? Am I understanding correctly?

No
Lite-XL wants iconv IF its not an extra dependency.
Otherwise this would require iconv.dll on Windows and libiconv.so on BSD.

Reply:
So unconditionally enable it on Linux and unconditionally disable it everywhere else?

Reply:
not necessarily.
iconv being included in libc is not a given and is just something glibc does and musl decided to copy for compatibility.
Other libcs that may be used on Linux may not implement it.

The idea is to seperate between toolchain iconv and library iconv.

Reply:
We don't do that for libm or libdl, even though both are more likely to require a shared library than iconv is -- e.g. it's a shared library on glibc.

What if you have libiconv.a installed, which is suitable for not having to require libiconv.so or iconv.dll?

>  Other libcs that may be used on Linux may not implement it.

Which libcs for Linux are we talking about here? We're not exactly spoiled for choice -- a libc is a complex endeavor that, unless being statically linked -- in which case none of this discussion even matters, since you *have* to statically link iconv too -- requires extensive platform support, you need to compile your entire distro for it, or run multiple ***system libcs***.

Your choices are exactly glibc and musl.

Even if you wanted to statically link against a private libc, you don't really have a lot of choices when it comes to a libc. Every remaining option specifically targets embedded devices, bionic is not actually used outside of Android, dietlibc doesn't even support shared libraries, if you say "klibc" I'm just going to laugh, and...

... all of them other than klibc *still* support iconv as a builtin, not as an external library. For uclibc you might need to enable iconv support at configure time.

Reply:
Soooo.

I reiterate: when building redistributables, unconditionally enable it on Linux, unconditionally disable it everywhere else.

Reply:
#882